File: src/main/java/com/kishlaly/ta/analyze/tasks/blocks/groups/ThreeDisplays_Sell_1.java
package com.kishlaly.ta.analyze.tasks.blocks.groups;

import com.kishlaly.ta.analyze.tasks.blocks.TaskBlockJava;
import com.kishlaly.ta.analyze.tasks.blocks.commons.ScreenBasicValidationJava;
import com.kishlaly.ta.analyze.tasks.blocks.one.Short_ScreenOne_StrictTrendCheckJava;
import com.kishlaly.ta.analyze.tasks.blocks.two.*;

import java.util.ArrayList;
import java.util.List;

public class ThreeDisplays_Sell_1 implements BlocksGroupJava {

    public List<TaskBlockJava> blocks() {
        return new ArrayList<TaskBlockJava>() {{
            add(new ScreenBasicValidationJava());
            add(new Short_ScreenOne_StrictTrendCheckJava());

            add(new Short_ScreenTwo_MACD_ThreeAboveZeroAndDescending());
            add(new Short_ScreenTwo_Stoch_D_ThreeDescending());
            add(new Short_ScreenTwo_Stoch_D_K_ThreeDescendingFromOverbought());
            add(new Short_ScreenTwo_EMA_ThreeBarsDescendingAndCrossing());
            add(new Short_ScreenTwo_EMA_LastBarTooLow());
        }};
    }

    @Override
    public String comments() {
        return "The first short strategy. Not stable";
    }
}

File: src/main/kotlin/com/kishlaly/ta/analyze/tasks/groups/threedisplays/ThreeDisplays_Sell_1.kt
package com.kishlaly.ta.analyze.tasks.groups.threedisplays

import com.kishlaly.ta.analyze.tasks.blocks.TaskBlock
import com.kishlaly.ta.analyze.tasks.blocks.commons.ScreenBasicValidation
import com.kishlaly.ta.analyze.tasks.blocks.one.Short_ScreenOne_StrictTrendCheck
import com.kishlaly.ta.analyze.tasks.groups.AbstractBlocksGroup

class ThreeDisplays_Sell_1 : AbstractBlocksGroup() {

    override fun blocks(): List<TaskBlock> {
        return object : ArrayList<TaskBlock>() {
            init {
                add(ScreenBasicValidation())
                add(Short_ScreenOne_StrictTrendCheck())
//                add(Short_ScreenTwo_MACD_ThreeAboveZeroAndDescending())
//                add(Short_ScreenTwo_Stoch_D_ThreeDescending())
//                add(Short_ScreenTwo_Stoch_D_K_ThreeDescendingFromOverbought())
//                add(Short_ScreenTwo_EMA_ThreeBarsDescendingAndCrossing())
//                add(Short_ScreenTwo_EMA_LastBarTooLow())
            }
        }
    }

    override fun comments() = super.comments() + "The first short strategy. Not stable"

}